src/Components/NewsletterUpdate/NewsletterUpdateContainer.js

```diff
@@ -0,0 +1,201 @@
+            newsletters: action.payload.newsletters,
+            didSubToNewslettersBefore:
+              action.payload.didSubToNewslettersBefore,
+            isListLoading: false
+          });
+
+        case SET_SELECT:
+          return Update({
+            ...state,
+            newsletters: state.newsletters.map((newsletter) => {
+              if (newsletter.id === action.payload) {
+                return {
+                  ...newsletter,
+                  selected: !newsletter.selected
+                };
+              }
+              return newsletter;
+            })
+          });
+
+        case SWITCH_TO_UPDATE:
+          return Update({
+            ...state,
+            didSubToNewslettersBefore: true
+          });
+
+        case SHOW_ALERT:
+          return Update({
+            ...state,
+            alert: action.payload
+          });
+
+        case LOADING:
+          return Update({ ...state, isSubmitting: action.payload });
+
+        case HANDLE_SUBMIT:
+          return UpdateWithSideEffect(
+            { ...state, isSubmitting: true },
+            (state, dispatch) => handleSubmit(state, dispatch)
+          );
+
+        default:
+          return state;
+      }
+    },
+    initialState
+  );
+
+  useEffect(() => {
+    const getUserNewsletters = () => {
+      dispatch({
+        type: GET_NEWSLETTERS_FETCH
+      });
+
+      window.Pelcro.newsletter.getByEmail(
+        window.Pelcro.user.read()?.email,
+        (err, res) => {
+          if (err) {
+            return dispatch({
+              type: SHOW_ALERT,
+              payload: {
+                type: "error",
+                content: getErrorMessages(err)
+              }
+            });
+          }
+
+          const newsletters =
+            window.Pelcro?.uiSettings?.newsletters ?? [];
+          const selectedNewsletters =
+            res.data.lists?.split(",") ?? [];
+          const allNewslettersWithSelectedField = newsletters.map(
+            (newsletter) => ({
+              ...newsletter,
+              id: String(newsletter.id),
+              selected: selectedNewsletters.includes(
+                String(newsletter.id)
+              )
+            })
+          );
+          dispatch({
+            type: GET_NEWSLETTERS_SUCCESS,
+            payload: {
+              newsletters: allNewslettersWithSelectedField,
+              didSubToNewslettersBefore: Boolean(res.data.email)
+            }
+          });
+        }
+      );
+    };
+
+    getUserNewsletters();
+  }, []);
+
+  return (
+    <div
+      style={{ ...style }}
+      className={`pelcro-container pelcro-newsletter-update-container ${className}`}
+    >
+      <Provider value={{ state, dispatch }}>
+        {children.length
+          ? children.map((child, i) => {
+              if (child) {
+                return React.cloneElement(child, { store, key: i });
+              }
+            })
+          : React.cloneElement(children, { store })}
```

Comment:
Aren't we now using the React.children utility? Where there a problem with it?

Reply:
will update it